src/hotspot/share/opto/phaseX.cpp

```diff
@@ -62,6 +62,31 @@ NodeHash::NodeHash(Arena *arena, uint est_max_size) :
   memset(_table,0,sizeof(Node*)*_max);
 }
 
+//-----------------------------------------------------------------------------
+
+bool NodeHash::check_for_collision(const Node* n, const Node* k) const {
```

Comment:
```suggestion
bool NodeHash::have_equivalent_inputs(const Node* n, const Node* k) const {
```
I think that would be a bit more explicit.